src/sage/rings/function_field/drinfeld_modules/charzero_drinfeld_module.py

```diff
@@ -228,7 +238,13 @@ def coeff_exp(k):
                 return self._compute_coefficient_exp(v)
             else:
                 return zero
-        return L(coeff_exp, valuation=1)
+
+        if prec is Infinity:
+            L = LazyPowerSeriesRing(self._base, name)
+            return L(coeff_exp, valuation=1)
+        else:
+            L = PowerSeriesRing(self._base, name, default_prec=prec)
+            return L([0] + [coeff_exp(i) for i in range(1,prec)], prec=prec)
```

Comment:
```suggestion
        L = PowerSeriesRing(self._base, name, default_prec=prec)
        return L([0] + [coeff_exp(i) for i in range(1,prec)], prec=prec)
```
(and actually also in `coeff_exp`)